### ISSUE TITLE
Docs: Improve accessibility of current (Sphinx classic) theme by adding lang

### ISSUE THREAD

**Issue by bertiewooster** [sympy/sympy#22811]
Set the language to English to ensure visitors using screen readers have the text announced correctly. This is a simple improvement that can be made while we work on using a new theme in #22716.

From Lighthouse:

> If a page doesn't specify a lang attribute, a screen reader assumes that the page is in the default language that the user chose when setting up the screen reader. If the page isn't actually in the default language, then the screen reader might not announce the page's text correctly. Learn more.
